src/Providers/AbstractServiceProvider.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Jjr\RobustChat\Providers;
+
+use Jjr\RobustChat\Configuration;
+use Jjr\RobustChat\Contracts\ServiceProviderInterface;
+use Jjr\RobustChat\DTO\Response\Response;
+
+/**
+ * Abstract class implementing common functionality for AI service providers.
+ */
+abstract class AbstractServiceProvider implements ServiceProviderInterface
+{
+    /**
+     * AbstractServiceProvider constructor.
```

Comment:
@aghorianducalis In general, I'd like avoid a comment like this because it is clear from the function __construct about it's main role.
Just add a comment when it clarifies what is happening, or is a note to other devs about the state of the code